### DISCUSSION TITLE
API to manage repo secrets?

### DISCUSSION THREAD

**Discussion by j3parker** [community/community#25023] [Actions]
<p>Is there going to be an API to manage (create/update/delete) repository secrets?</p>
<p>Our use-case is for rotating AWS IAM credentials automatically. To deploy to AWS we need IAM credentials in GitHub actions, but we have strict requirements about how IAM credentials are used at my organization (they can only be used by external services like GitHub actions or Circle CI, and they must be automatically rotated). So this is blocking us from adopting GitHub actions broadly.</p>
<p>I don’t see any documented API yet - are there plans to release one?</p>
<p>Thanks.</p>

**Comment by j3parker** [comment, marked as answer]
<p>Found a related question: <a href="https://github.community/t/github-apps-to-add-secrets/16124">https://github.community/t5/GitHub-Actions/Github-Apps-to-add-secrets/m-p/28259</a></p>

**Comment by chrispat** [comment]
<p>The Actions api shipped today <a href="https://developer.github.com/v3/actions/" rel="nofollow noopener">https://developer.github.com/v3/actions/</a></p>

**Comment by mpdude** [comment]
<p><a class="mention" href="/chrispat">@chrispat</a> Could you relay a question/message to your support team?</p>
<p>I’ve played with the API and found that the <code>GET /repos/:owner/:repo/actions/secrets</code> endpoint returns timestamps like <code>2020-01-29T13:19:12Z</code>. However, it seems those are not really UTC times but rather in the UTC-8 time zone.</p>
<p>Could they please check that?</p>

**Comment by chrispat** [comment]
<p>Thanks for letting me know.  I have passed the info over to the engineering team.</p>

**Comment by zhuzeng** [comment]
<p>This is great news! May I ask when the go-github client library will be updated to include the functionalities? <a href="https://godoc.org/github.com/google/go-github/github" rel="nofollow noopener">https://godoc.org/github.com/google/go-github/github</a></p>
